README.md

```diff
@@ -1,3 +1,1030 @@
+        "search/search.html",
+        {
+            "search_query": search_query,
+            "search_results": search_results,
+        },
+    )
+```
+
+We can make a few nice-to-have customizations to the template which displays the results. First, add `| <a href="/search/">Search</a>` to our `header.html` as the last menu item, so all our pages link to the search form.
+
+Then, in `search/templates/search/search.html`, we will:
+
+- Convert the `<ul>` list to `<ol>` for better semantics – search results are ordered by relevance.
+- Add a message at the top of the results to summarize the query and number of results.
+- Add page counts to the pagination at the bottom.
+
+Summary before the `<ol></ol>` results list:
+
+```html
+<p>You searched{% if search_query %} for “{{ search_query }}”{% endif %}, {{ search_results.paginator.count }} result{{ search_results.paginator.count|pluralize }} found.</p>
+```
+
+Page counts after the `<ol></ol>` results list:
+
+```html
+{% if search_results.paginator.num_pages > 1 %}
+    <p>Page {{ search_results.number }} of {{ search_results.paginator.num_pages }}, showing {{ search_results|length }} result{{ search_results|pluralize }} out of {{ search_results.paginator.count }}</p>
+{% endif %}
+```
+
+#### Make more page content searchable
+
+TODO. Sample code:
+
+```python
+from wagtail.search import index
+
+    search_fields = Page.search_fields + [
+        index.SearchField("body"),
+    ]
+```
+
+### Deployment
+
+- Fly.io setup
+- Django settings
+- Database
+- Media files
+- Dockerfile
+
+### Optional steps
+
+#### Convert BlogPage to use StreamField
+
+#### Organise the Wagtail admin with a SnippetViewSetGroup
+
+Commit: [56c32fa](https://github.com/thibaudcolas/your-wagtail-portfolio/commit/56c32fa) Register models with a SnippetViewSetGroup
+
+- Not necessarily the most important thing, just demonstrates more Wagtail features. Perhaps later in the tutorial with other Wagtail customizations?
+- Showcases `wagtail_hooks.py`
+
+Remove `@register_snippet` on FooterText model, then proceed.
+
+New file: `base/wagtail_hooks.py`
+
+```python
+
+from wagtail.snippets.models import register_snippet
+from wagtail.snippets.views.snippets import SnippetViewSet, SnippetViewSetGroup
+
+from base.models import FooterText
+from blog.models import Author
+
+
+class FooterTextViewSet(SnippetViewSet):
+    model = FooterText
+    icon = "pilcrow"
+    search_fields = ("body",)
+
+
+class AuthorViewSet(SnippetViewSet):
+    model = Author
+    icon = "user"
+
+
+class SiteDataSnippetViewSetGroup(SnippetViewSetGroup):
+    menu_label = "Site Data"
+    menu_icon = "info-circle"  # change as required
+    menu_order = 300  # will put in 4th place (000 being 1st, 100 2nd)
+    items = (FooterTextViewSet, AuthorViewSet)
+
+register_snippet(SiteDataSnippetViewSetGroup)
+```
+
+
+
+## Potential tutorial additions
+
+Solid options:
+
+```
+Django Debug Toolbar
+Sitemap
+parent_page_types and subpage_types
```

Comment:
I personally believe these two subjects are important. Wagtail has the unique ability to create websites that can't be broken by content editors - and keeping pages in the right places in the tree is part of that. But that's just my 2 cents.